activerecord/lib/active_record/schema_dumper.rb

```diff
@@ -243,7 +243,9 @@ def format_options(options)
       end
 
       def remove_prefix_and_suffix(table)
-        table.gsub(/^(#{@options[:table_name_prefix]})(.+)(#{@options[:table_name_suffix]})$/,  "\\2")
+        prefix = Regexp.escape(@options[:table_name_prefix].to_s)
+        suffix = Regexp.escape(@options[:table_name_suffix].to_s)
+        table.sub(/\A(?:#{prefix})(.+)(?:#{suffix})\z/, "\\1")
```

Comment:
Why the non-capturing groups around the prefix/suffix?  Given the escaping, do they accomplish anything?

I generally like the change.